base/dataset.go

```diff
@@ -110,6 +110,15 @@ func ListDatasets(r repo.Repo, term string, limit, offset int, RPC, publishedOnl
 		}
 		res = pub[:i]
 	}
+	// if offset is too high, return empty list
+	if offset >= len(res) {
+		return res[:0], nil
```

Comment:
This reads a bit strange - using the name `res` but essentially ignoring its value. Would suggest changing it to: `		return []repo.DatasetRef{}, nil`